modules/ismp/testsuite/src/lib.rs

```diff
@@ -402,7 +434,188 @@ pub fn check_request_source_and_destination() {}
 
 /// This should check that if a proxy isn't configured, responses are not valid if they don't come
 /// from the state machine claimed in the proof
-pub fn check_response_source() {}
+pub fn check_response_source<H, D>(host: &H, dispatcher: &D) -> Result<(), &'static str>
```

Comment:
Test is in the wrong direction.
First assert that host.allowed_proxy().is_none().
Then try to handle a request and response whose destinatin chain != host.state_machine()
Assert that the request or response is ignored.